### PR TITLE
Fixes re-loading job data on start_scan tube

### PR DESCRIPTION
**Pull request by navidshaikh**
Prevents re-scanning already scanned container and streamlines orchestration to poll_server tube.

### CODE REVIEW

File: scanning/scanners/runner.py [hunk 1]
```diff
@@ -333,7 +332,8 @@ def scan(self):
         # case where this job is running after jenkins polling jobs
         # we need to run only one scanner; analytics-integration with
         # scan_type=report
-        if self.job.get("gemini_report"):
```
Comment (navidshaikh) [inline scanning/scanners/runner.py:336]: this was the bug, the condition was failing for cases
 - `gemini_report` not present in `self.job`
- `gemini_report=False` for cases where gemini server doesn't return the reports.

File: workers/scan.py [hunk 2]
```diff
@@ -26,40 +26,37 @@ def handle_job(self, job):
...
-        poll_job = self.job.copy()
-        # now put job for polling
-        self.put_job_for_polling(poll_job)
+        # presence of `gemini_report` hints if polling is done or not
+        # following will be executed after container is registered
+        if "gemini_report" not in job:
+            # copy the job in another var to avoid action var overwrite
+            poll_job = job.copy()
+            # here there is another clone of job params loaded on beanstalkd
+            self.put_job_for_polling(poll_job)
```
Comment (bamachrn) [inline workers/scan.py:54]: are we not changing the action before putting the job back to beanstalkd?

File: scripts/weeklyscan.py [hunk 3]
```diff
@@ -153,7 +153,7 @@ def put_image_for_scanning(self, image, logs_dir, giturl, gitsha):
         }
         self.logger.info("Putting {} for scan..".format(image))
         # now put image for scan
-        self.queue.put(json.dumps(job))
+        self.queue.put(json.dumps(job), "master_tube")
```
Comment (bamachrn) [inline scripts/weeklyscan.py:156]: :+1:

File: workers/scan.py [hunk 4]
```diff
@@ -26,40 +26,37 @@ def handle_job(self, job):
...
-        scanners_data.pop("scan_results", None)
 
-        self.logger.info("Putting job for polling at poll_server tube.")
-        # copy the job to another variable to avoid "action" overriding
-        poll_job = self.job.copy()
-        # now put job for polling
-        self.put_job_for_polling(poll_job)
+        # presence of `gemini_report` hints if polling is done or not
+        # following will be executed after container is registered
+        if "gemini_report" not in job:
```
Comment (navidshaikh) [inline workers/scan.py:50]: and this should fix the orchestration while we put job to `poll_server` tube.

**Review by bamachrn — CHANGES_REQUESTED** [review]

**Review by bamachrn — APPROVED** [review]

File: workers/scan.py (hunk 2 above)
Comment (navidshaikh) [reply workers/scan.py:54]: Nope, we are not changing the action. As we are directly putting the job to `poll_server_tube`.

**Pull request merged**